### PR TITLE
Implement `dpnp.isneginf` and `dpnp.isposinf`

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR suggests adding 2 logical elementt-wise functions `dpnp.isposinf()` and `dpnp.isneginf()` by reusing `dpnp.inf()` and `dpnp.signbit()` and new tests to cover them

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_logic.py [hunk 1]
```diff
@@ -710,6 +712,134 @@ def isclose(x1, x2, rtol=1e-05, atol=1e-08, equal_nan=False):
...
+    >>> y = np.zeros(x.shape, dtype='bool')
+    >>> np.isneginf(x, y)
+    array([ True, False, False])
+    >>> y
+    array([ True, False, False])
+
+    """
+
+    is_inf = dpnp.isinf(x)
+    try:
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_logic.py:767]: Wouldn't it be better to check for complex type and raise `TypeError`? Or is other other use cases when `dpnp.signbit` can raise an exception?

File: dpnp/dpnp_iface_logic.py [hunk 2]
```diff
@@ -710,6 +712,134 @@ def isclose(x1, x2, rtol=1e-05, atol=1e-08, equal_nan=False):
...
+    For full documentation refer to :obj:`numpy.isneginf`.
+
+    Parameters
+    ----------
+    x : {dpnp.ndarray, usm_ndarray}
+        Input array.
+    out : {None, dpnp.ndarray, usm_ndarray}, optional
+        A location into which the result is stored. If provided, it must have a
+        shape that the input broadcasts to and a boolean data type.
+        If not provided or None, a freshly-allocated boolean array is returned
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_logic.py:728]:
```suggestion
        If not provided or ``None``, a freshly-allocated boolean array is returned.
        Default: ``None``.
```

File: dpnp/dpnp_iface_logic.py [hunk 3]
```diff
@@ -710,6 +712,134 @@ def isclose(x1, x2, rtol=1e-05, atol=1e-08, equal_nan=False):
 )
 
 
+def isneginf(x, out=None):
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_logic.py:715]: `isneginf` and `isposinf` are missing in generated documentation

File: dpnp/dpnp_iface_logic.py [hunk 4]
```diff
@@ -710,6 +712,134 @@ def isclose(x1, x2, rtol=1e-05, atol=1e-08, equal_nan=False):
...
+    try:
+        signbit = dpnp.signbit(x)
+    except ValueError as e:
+        dtype = x.dtype
+        raise TypeError(
+            f"This operation is not supported for {dtype} values "
+            "because it would be ambiguous."
+        ) from e
+
+    return dpnp.logical_and(is_inf, signbit, out)
```
Comment (antonwolfy) [inline dpnp/dpnp_iface_logic.py:776]:
```suggestion
    return dpnp.logical_and(is_inf, signbit, out=out)
```

File: tests/test_logic.py [hunk 5]
```diff
@@ -394,3 +393,32 @@ def test_finite(op, data, dtype):
     dpnp_res = getattr(dpnp, op)(x, out=dp_out)
     assert dp_out is dpnp_res
     assert_equal(dpnp_res, np_res)
```
Comment (antonwolfy) [inline tests/test_logic.py:395]: What is about tests in SYCL queue and USM type scopes?

File: tests/test_logic.py [hunk 6]
```diff
@@ -394,3 +393,32 @@ def test_finite(op, data, dtype):
...
+    ids=[
+        "1D array",
+        "2D array",
+    ],
+)
+@pytest.mark.parametrize("dtype", get_float_complex_dtypes())
+def test_infinity_sign(func, data, dtype):
+    x = dpnp.asarray(data, dtype=dtype)
+    if dpnp.issubdtype(dtype, dpnp.complexfloating):
+        with pytest.raises(TypeError):
```
Comment (antonwolfy) [inline tests/test_logic.py:414]: I would propose to split into two tests. In the 1st one we will need to check that numpy and dpnp both raises the same exception.

File: tests/test_logic.py (hunk 5 above)
Comment (vlad-perevezentsev) [reply tests/test_logic.py:396]: I have added new tests for logic functions
Thank you!

File: tests/test_sycl_queue.py [hunk 7]
```diff
@@ -478,6 +478,15 @@ def test_meshgrid(device_x, device_y):
         pytest.param("trapz", [[0.0, 1.0, 2.0], [3.0, 4.0, 5.0]]),
         pytest.param("trunc", [-1.7, -1.5, -0.2, 0.2, 1.5, 1.7, 2.0]),
         pytest.param("var", [1.0, 2.0, 4.0, 7.0]),
+        # logic functions
+        pytest.param("all", [-dpnp.inf, -1.0, 1.0, dpnp.inf, dpnp.nan]),
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply tests/test_sycl_queue.py]: I have added separate tests for logic functions

File: tests/test_usm_type.py [hunk 8]
```diff
@@ -354,6 +354,37 @@ def test_tril_triu(func, usm_type):
     assert x.usm_type == usm_type
 
 
+@pytest.mark.parametrize(
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply tests/test_usm_type.py:357]: In order not to expand the already huge test I added new tests as in `test_usm_type.py`

File: dpnp/dpnp_iface_logic.py [hunk 9]
```diff
@@ -710,6 +712,158 @@ def isclose(x1, x2, rtol=1e-05, atol=1e-08, equal_nan=False):
...
+    """
+
+    dpnp.check_supported_arrays_type(x)
+
+    if out is not None:
+        dpnp.check_supported_arrays_type(out)
+        out_dtype = out.dtype
+        if not dpnp.issubdtype(out_dtype, dpnp.bool):
+            raise TypeError(
+                f"Output array data type must be boolean, got {out.dtype}"
```
[the remark this answers is not in the document]
Comment (antonwolfy) [reply dpnp/dpnp_iface_logic.py]: It would be great to create an issue to track that.

**Review by antonwolfy — APPROVED** [review]
Thank you @vlad-perevezentsev

File: dpnp/dpnp_iface_logic.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply dpnp/dpnp_iface_logic.py]: I have created an issue to track that [#1717](https://github.com/IntelPython/dpctl/issues/1717) and added a comment line with it to code

**Review by antonwolfy — APPROVED** [review]

**Pull request merged**